sycl/test-e2e/README.md

```diff
@@ -379,3 +379,103 @@ Note: please avoid using `REQUIRES: TEMPORARY_DISABLED` for this purpose, it's
 a non-standard mechanism. Use `UNSUPPORTED: true` instead, we track
 `UNSUPPORTED` tests using the mechanism described above. Otherwise the test
 risks remaining untraceable.
+
+## Compiling and executing tests on separate systems
+
+The execution of e2e tests can be separated into compilation and execution
+stages via the `test-mode` lit parameter. This allows us to reduce testing time
+by compiling tests on more powerful systems and reusing the binaries on other
+machines. By default the `test-mode` parameter is set to `full`, indicating
+that both stages will run. This parameter can be set to `build-only`, or
+`run-only`, to only run the compilation stage, or the execution stage
+respectively.
+
+The modes work as follow:
+### `--param test-mode=full`
+This is the default mode tests run in. Tests are marked as unsupported if no
+device on the machine can fulfill the `REQUIRES`/`UNSUPPORTED` statements. In
```

Comment:
Add something about improving that logic in the future.

Reply:
> > NOTE: currently a WIP to make this mode properly react to features
> 
> Can we drop WIP stuff from this PR and add it to the doc once it is implemented? I personally prefer to have docs in sync with the current implementation.

I originally removed such comments because of @uditagarwal97's review. I also think it would be good to add so that people do not think that this is a permanent limitation.

Reply:
> I also think it would be good to add so that people do not think that this is a permanent limitation.

In that case, you can add a separate section on WIP and add it there. That way, it would provide a clear distinction between what's currently supported and what's tabled for the future (which may or may not manifest).

Reply:
looks good.